app/models/character.rb

```diff
@@ -31,4 +31,42 @@ def assign_skills
       skills.create(json_skill.attributes)
     end
   end
+
+  def upgrade_skills(skills_names)
+    skills.where(name: skills_names).update_all("rank = rank + 1")
+  end
+
+  def verify_skill(skill, status)
+    if (self.career_skill?(skill.name))
+      self.set_rank(skill, skill.career_skill, status)
+    else 
+      self.set_rank(skill, skill.normal_skill, status)
+    end
+  end
+
+  def set_rank(skill, skill_xp, status)
```

Comment:
You have a method argument dependency here...don't feel quite sure about it. I think it has to be with the `verify_skill` method...probable you need to change that...or move some logic to the skill model